openedx/openedx-platform
# [FC-0009] [FAL-3496] Copy/paste unit from within a unit

## Description

This PR adds the option to copy/paste a unit from within a unit in CMS as per the designs in [this wireframe](https://www.figma.com/file/y2qfzidTxxfYK8ONgyT7KP/Copy-Paste-Wireframes?type=design&node-id=3104-6457&mode=design&t=zE4OMczgdKsB1573-0). It also removes the `last published` section.

## Supporting Information
https://github.com/openedx/modular-learning/issues/96


OpenCraft Internal Jira task - [FAL-3496](https://tasks.opencraft.com/browse/FAL-3496)

## Screenshots

1. Copy button section:

![copy-unit-button](https://github.com/openedx/edx-platform/assets/13742492/b743726e-9d1c-4ee6-b31f-c62b509b9217)


2. Copy button hover/select:

![copy-unit-button-selected](https://github.com/openedx/edx-platform/assets/13742492/4022aebc-ad7c-49b1-b29c-4e6eead45184)



3. Dropdown menu options when unit is copied:

![Screenshot from 2023-11-30 16-53-43](https://github.com/openedx/edx-platform/assets/13742492/2f8ba999-de57-4c59-97ab-203905809ed8)


## Testing Instructions
1. Checkout the PR branch on the devstack.
2. Enable the `contentstore.enable_copy_paste_units` waffle flag in cms admin panel.
3. In Studio, open a unit and verify that the dropdown that shows up when `New Unit` button is clicked, only contains one option i.e. `New Unit`
4. Click on `Copy to Clipboard` button.
5. Verify that the dropdown now contains another option `Paste as new unit`.
6. Select `Paste as new unit` option and verify that the copies unit is pasted as a new unit.
7. Copy only a component from the unit to clipboard and verify that the paste option doesn't show up.
